### PR TITLE
Web: Update puppeteer to 12.0.1

### PR DESCRIPTION
**Pull request by vovakulikov**
This PR simply updates the puppeteer version to 12.0.1. At the first glance, it should be an easy update since the puppeteer types package hasn't been changed. 

Most problems came from puppeteer types and their fixes don't change anything in logic dramatically.

### CODE REVIEW

**Comment by vovakulikov** [comment]
Hi @tjkandala. I'm not sure about changes and CDP adapter and in browser integration tests. Whenever you got time can you please check browser extension tests

File: client/shared/scripts/download-puppeteer-browser.ts [hunk 1]
```diff
@@ -9,7 +9,12 @@ async function main(): Promise<void> {
         signale.error(`Puppeteer browser must be "chrome" or "firefox", but got: "${browserName}"`)
         process.exit(1)
     }
-    const browserFetcher = puppeteer.createBrowserFetcher({ product: browserName })
+
+    // Current puppeteer version (12.0.1) has a broken types
+    // See https://github.com/puppeteer/puppeteer/issues/7100
```
Comment (umpox) [inline client/shared/scripts/download-puppeteer-browser.ts:14]: 😢

**Review by umpox — APPROVED** [review]
Thank you!

**Review by valerybugakov — APPROVED** [review]
Thank you! LGTM

**Review by tjkandala — APPROVED** [review]
I ran the browser extension integration tests and they work, thanks!

**Pull request merged**